### ISSUE TITLE
implement the feature to produce the waves using phone audio jack

### ISSUE THREAD

**Issue by Avjeet** [fossasia/pslab-android#1165]
**Actual Behaviour**

Currently their is no such functionality available to produce different from audio jack of phone

**Expected Behaviour**

It is an additional functionality,  in this, the audio jack of the phone can be used to produce different waves like sine, square, triangular which can be analyzed in the oscilloscope. 

**Steps to reproduce it**

Wave generator -> Implement the functionality using AudioTrack class provided by android 

**Would you like to work on the issue?**

Yes

**Comment by harsh-2711** [comment]
@Avjeet It would be great if you first focus on the digital waves generator pins as it's difficult for me to everytime use `Control Activity` for testing and then remove the changes before commit. No digital pins i.e. SQR1, SQR2, SQR3, and SQR4 are working correctly.

**Comment by ho-dor** [comment]
I would like to work on this. 
Reference :- [https://blog.fossasia.org/filling-audio-buffer-to-generate-waves-in-the-pslab-android-app/](url)

**Comment by ho-dor** [comment]
@neel1998 @CloudyPadmal @mariobehling I don't think this is any different from the in-built mic functionality that we have already as we are using AudioJack instance for that functionality as well. This can be closed.

**Comment by neel1998** [comment]
@CloudyPadmal @mariobehling  please close this

**Comment by neel1998** [comment]
@CloudyPadmal  @mariobehling can we close this?

**Comment by CloudyPadmal** [comment]
This is about producing waves; not observing. Do we have that feature?

**Comment by neel1998** [comment]
I didn't understand your point. Currently we are displaying the wave produced by the sound on the oscilloscope. What do you mean by producing...??

**Comment by CloudyPadmal** [comment]
Can you read and understand the **title** of this issue? You can also read the issue body if it's still hard for you to understand.

**Comment by neel1998** [comment]
@CloudyPadmal here is what i hve understood, the user will decide some wave let's say some sine/triangular wave with freq,amp and phase, and we are suppose to produce a sound wave based on that. For example, whistling is a kind of a sine wave, so if user gives sine wave to this feature, a sound like whistle should be produced. Am i correct?

**Comment by CloudyPadmal** [comment]
https://blog.fossasia.org/filling-audio-buffer-to-generate-waves-in-the-pslab-android-app/
https://blog.fossasia.org/using-the-audio-jack-to-make-an-oscilloscope-in-the-pslab-android-app/
